### PR TITLE
adds codewhisperer 0.12.0 cask

### PR DESCRIPTION
**Pull request by ajzeigert**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/c/codewhisperer.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+cask "codewhisperer" do
+  version "0.12.0"
+  sha256 :no_check
+
+  url "https://desktop-release.codewhisperer.us-east-1.amazonaws.com/latest/CodeWhisperer.dmg", verified: "desktop-release.codewhisperer.us-east-1.amazonaws.com/"
+  name "CodeWhisperer for Command Line"
+  desc "IDE-style autocomplete for 500+ CLIs. Translate natural language-to-bash"
+  homepage "https://docs.aws.amazon.com/codewhisperer/latest/userguide/command-line-getting-started-installing.html"
```
Comment (bevanjkay) [inline Casks/c/codewhisperer.rb:8]:
```suggestion
  homepage "https://aws.amazon.com/codewhisperer/"
```

File: Casks/c/codewhisperer.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+cask "codewhisperer" do
+  version "0.12.0"
+  sha256 :no_check
+
+  url "https://desktop-release.codewhisperer.us-east-1.amazonaws.com/latest/CodeWhisperer.dmg", verified: "desktop-release.codewhisperer.us-east-1.amazonaws.com/"
+  name "CodeWhisperer for Command Line"
+  desc "IDE-style autocomplete for 500+ CLIs. Translate natural language-to-bash"
```
Comment (bevanjkay) [inline Casks/c/codewhisperer.rb:7]:
```suggestion
  desc "AI-powered productivity tool for the IDE and command line"
```

**Review by bevanjkay — CHANGES_REQUESTED** [review]
@ajzeigert I have added a `livecheck` stanza and updated the `url` to be versioned to match. Please see additional review comments.

File: Casks/c/codewhisperer.rb (hunk 2 above)
Comment (ajzeigert) [reply Casks/c/codewhisperer.rb:8]: This cask is specifically for the command line tool, which is separate from the IDE plugins. The description I provided was taken directly from how the app describes itself, but I understand if that is too long.

Screenshot 2023-11-28 at 10 24 57 AM

Screenshot 2023-11-28 at 10 25 02 AM

Maybe just "AI-powered productivity tool for the command line"?

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**